Sources/main.cpp

```diff
@@ -490,16 +557,7 @@ static int VersionMain(int argc, char **argv) {
 __declspec(dllexport)
 #endif
 int main(int argc, char **argv) {
-// clang-format on
-
-#if defined(OS_POSIX)
```

Comment:
I think you should just move this block below, after the argument parsing is done, and add a simple `if ()` statement in there to avoid closing the file descriptor you get passed by the `--fd` argument.

Reply:
But options are defined and parsed in the sub-mode's main functions. We don't know have the fd info in main(). I want to postpone the close only for g mode. That is why I changed the switch()

Reply:
Good point. Let me think about this for a second.